src/refcount.h

```diff
@@ -1,10 +1,33 @@
-#include <stdatomic.h>
+#ifndef _MSC_VER
+#  include <stdatomic.h>
 
 typedef atomic_size_t Refcount;
 
-#define refcount_inited(counter) (refcount_load(counter) != 0)
-#define refcount_load(counter) atomic_load(counter)
-#define refcount_init(counter, value) atomic_init(counter, value)
-#define refcount_inc(counter) atomic_fetch_add_explicit(counter, 1, memory_order_relaxed)
-#define refcount_dec(counter) atomic_fetch_sub_explicit(counter, 1, memory_order_acq_rel)
-#define refcount_destroy(count) ((void)0)
+#  define refcount_inited(counter) (refcount_load(counter) != 0)
+#  define refcount_load(counter) atomic_load(counter)
+#  define refcount_init(counter, value) atomic_init(counter, value)
+#  define refcount_inc(counter) atomic_fetch_add_explicit(counter, 1, memory_order_relaxed)
+#  define refcount_dec(counter) atomic_fetch_sub_explicit(counter, 1, memory_order_acq_rel)
+#  define refcount_destroy(count) ((void)0)
+#else
+/* Visual C++ doesn't support C11 atomics yet. However, Windows documentation
+ * guarantees that simple reads and writes are atomic:
+ * https://docs.microsoft.com/en-us/windows/win32/sync/interlocked-variable-access
+ */
+#  include <windows.h>
+
+typedef volatile size_t Refcount;
+
+#  define refcount_inited(counter) (*(counter) != 0)
```

Comment:
Wouldn't using counter as a pointer require it to be allocated first? This looks weird to me.

Reply:
This is exactly the same logic as in the C11 version. The code calls refcount_inited like this:
```c
if (!refcount_inited(&thread_counter)) {
```

Reply:
Right, I forgot about that.